s3err/s3err.go

```diff
@@ -106,9 +106,9 @@ const (
 	ErrNotImplemented
 	ErrPreconditionFailed
 	ErrInvalidObjectState
-
 	ErrExistingObjectIsDirectory
 	ErrObjectParentIsFile
```

Comment:
these were separated out because they are non-standard AWS errors.  lets keep those separated just so we keep track of these. they probably could have used a comment indicating why they were separate.